dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/FeatureManager.java

```diff
@@ -1400,6 +1408,10 @@ protected boolean updateFeatures(WsLocationAdmin locService,
         provisioner.resolveBundles(bundleContext, installedBundles);
 
         if (featureChange.featureBundlesResolved != null) {
+            Map<String, Object> props = new HashMap<String, Object>(1);
```

Comment:
Does this map need to be thread safe?  It seems we populate it here and then leave it open to other threads to get/put stuff on it when the notification is fired.  Should setProperties just take the values and wrap them in an unmodifiable Map in the notification impl? This way they are only read-only on notification firing when they call getProperties.

Reply:
This has been addressed